### ISSUE TITLE
Peek doesn’t scale on mixed-dpi screens

### ISSUE THREAD

**Issue by fbruetting** [phw/peek#350]
Here you see that regular applications get scaled, when being dragged from the UHD screen to the HD one, but this doesn’t work for peek. Could this be fixed?

![peek 2018-08-11 12-36](https://user-images.githubusercontent.com/10520767/43990961-4ec8dfbe-9d63-11e8-84e7-00807102040d.gif)

**Comment by gort818** [comment]
I will work on this, just to make sure are you using X11 or Wayland?

**Comment by fbruetting** [comment]
Ah cool, I'm on Wayland.


Am 11.08.2018 8:46 nachm. schrieb Alessandro <notifications@github.com>:

I will work on this, just to make sure are you using X11 or Wayland?

**Comment by gort818** [comment]
Do know any other applications that work this way?  I would like to see how it is implemented.

**Comment by fbruetting** [comment]
I assume this could be due to some modifications which were done to GtkWindow or so, as regular GTK windows always scale like expected. There are just a few programs which don’t scale (Firefox and Electron applications). Maybe you find something in their bug trackers or can see if some GDK/GTK top levels were modified somehow?

**Comment by gort818** [comment]
Well I checked the tilix source code, I think I have a pretty good idea how it is done, will just need you to test it when it is ready :)

**Comment by fbruetting** [comment]
Yeah, should work automatically. Maybe it’s an issue due to changed DPI in conjunction with recording? Or do you use some kind of recording widget, that prevents DPI change? Maybe you can strip the code down to the very minimal basic window (commment out that video recording area), to see if that will work?

**Comment by gort818** [comment]
I don't have the hardware to test at the moment, can you see if the branch helps at all..not sure if it does anything at all.

https://github.com/gort818/peek/tree/hidpi
